crowbar_framework/app/controllers/barclamp_controller.rb

```diff
@@ -521,6 +521,7 @@ def valid_transition_states
       "applying", "discovered", "discovering", "hardware-installed",
       "hardware-installing", "hardware-updated", "hardware-updating",
       "installed", "installing", "ready", "readying", "recovering",
+      "crowbar_upgrade",
```

Comment:
I hate being a pain (and should have noticed faster), but the state itself can keep the dash as that's what is being used for other states too.

That being said, most of the other changes to underscore are fine and should be kept.